### PR TITLE
chore: provide a tool for generating config docs

### PR DESCRIPTION
**Pull request by neverchanje**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

To provide an automation tool that can help generate an up-to-date document for configurations. 
This tool does not require a separate command. Instead, it reuses `./risedev generate-example-config`.
The result can be copied without modification into our official doc upon every version release.

A follow-up task is to refine the rustdoc for important configurations, which are supposed to be maintained by developers instead of the doc team.

In addition, we may need to figure out an approach to extract the possible options for each config. We currently can only extract default values. For example,  there is no annoation to indicates `default_parallelism`'s other options except for 'full', 

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

https://github.com/risingwavelabs/risingwave-docs/issues/997

### CODE REVIEW

File: src/common/proc_macro/src/lib.rs [hunk 1]
```diff
@@ -263,3 +264,44 @@ pub fn session_config(input: TokenStream) -> TokenStream {
...
+/// that produces docs for each field.
+/// Unlike rustdoc, this tool focuses solely on extracting rustdoc for struct fields, without methods.
+///
+/// Example:
+///
+/// ```ignore
+/// #[derive(ConfigDoc)]
+/// pub struct Foo {
+///   /// Description for `a`.
+///   a i32,
```
Comment (BugenZhao) [inline src/common/proc_macro/src/lib.rs:278]:
```suggestion
///   a: i32,
```

Comment (BugenZhao) [inline src/common/proc_macro/src/lib.rs:278]:
```suggestion
///   a: i32,
```

**Review by BugenZhao — COMMENTED** [review]
Generally LGTM.

- Shall we apply the same ordering as `example.toml`?
- Should we include the developer configs?

**Review by xxchan — APPROVED** [review]
LGTM.

**Pull request merged**